server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -91,7 +91,6 @@
             NESTED_FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
             NESTED_FIELD_TYPE.setStored(true);
             NESTED_FIELD_TYPE.setOmitNorms(true);
-            NESTED_FIELD_TYPE.setStored(false);
```

Comment:
I think we want to remove the earlier `setStored` call rather than this one?  We only index the ID field in nested documents to ensure that they get deleted when their parent doc is deleted, we never load it from stored fields.

Reply:
Make sense, I have disabled id store field for nested docs.